src/core/include/openvino/op/util/avg_pool_base.hpp

```diff
@@ -0,0 +1,106 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/op/op.hpp"
+#include "openvino/op/util/attr_types.hpp"
+
+namespace ov {
+namespace op {
+namespace util {
+class OPENVINO_API AvgPoolBase : public Op {
+public:
+    OPENVINO_OP("AvgPool", "util");
+    AvgPoolBase() = default;
+
+    /// \brief      Constructs a batched average pooling operation.
+    ///
+    /// \param      arg            The output producing the input data batch tensor.<br>
+    ///                            `[d1, dn]`
+    /// \param      strides        The strides.<br> `[n]`
+    /// \param      pads_begin     The beginning of padding shape.<br> `[n]`
+    /// \param      pads_end       The end of padding shape.<br> `[n]`
+    /// \param      kernel         The kernel shape.<br> `[n]`
+    /// \param      exclude_pad    If false then averages include padding elements, each
+    ///                            treated as the number zero.  If true, padding
+    ///                            elements
+    ///                            are entirely ignored when computing averages.
+    /// \param      rounding_type  Whether to use ceiling or floor rounding type while
+    ///                            computing output shape.
+    /// \param      auto_pad       Padding type to use for additional padded dimensions
+    AvgPoolBase(const Output<Node>& arg,
+                const Strides& strides,
+                const Shape& pads_begin,
+                const Shape& pads_end,
+                const Shape& kernel,
+                bool exclude_pad,
+                op::RoundingType rounding_type = op::RoundingType::FLOOR,
+                const PadType& auto_pad = op::PadType::EXPLICIT);
+
+    void validate_and_infer_types() override;
+    bool visit_attributes(AttributeVisitor& visitor) override;
+
+    /// \return The kernel shape.
+    const Shape& get_kernel() const {
+        return m_kernel;
+    }
+    void set_kernel(const Shape& kernel) {
+        m_kernel = kernel;
+    }
+    /// \return The strides.
+    const Strides& get_strides() const {
+        return m_strides;
+    }
+    void set_strides(const Strides& strides) {
```

Comment:
Move implementation to cpp file.

Reply:
Moved in the commit https://github.com/openvinotoolkit/openvino/pull/23483/commits/f9b29236c83487e8e897157337e300ee4eaea169. Thanks.

`MaxPoolBase` has impls of getters/setters in the [header](https://github.com/openvinotoolkit/openvino/blob/master/src/core/include/openvino/op/util/max_pool_base.hpp) file. Should I move them to source file as well?

Reply:
@t-jankowski what do you think?

Reply:
This PR regards AvgPool, let's not mix the concerns.